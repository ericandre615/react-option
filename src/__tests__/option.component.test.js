import React from 'react';
import { render } from '@testing-library/react';

import {
    Some,
    None,
    Option,
} from '../index';

describe('Option Component', () => {
    it('should return children of Some when value prop evaluates to `true`', () => {
        const shouldRender = true;
        const { container: renderedOption, asFragment } = render(
            <Option value={ shouldRender }>
                <Some>
                    <div id="some">
                        <p>I am some content</p>
                    </div>
                </Some>
                <None>
                    <div id="none">
                        <p>I am none content</p>
                    </div>
                </None>
            </Option>
        )

        const renderedSome = renderedOption.querySelector('#some');
        const renderedNone = renderedOption.querySelector('#none');

        expect(renderedSome).toBeTruthy();
        expect(renderedNone).toBeFalsy();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should return children of None when value prop evaluates to `false`', () => {
        const shouldRender = false;
        const { container: renderedOption, asFragment } = render(
            <Option value={ shouldRender }>
              <Some>
                  <div id="some">
                      <p>I am some content</p>
                  </div>
              </Some>
              <None>
                  <div id="none">
                      <p>I am none content</p>
                  </div>
              </None>
            </Option>
        )

        const renderedSome = renderedOption.querySelector('#some');
        const renderedNone = renderedOption.querySelector('#none');

        expect(renderedSome).toBeFalsy();
        expect(renderedNone).toBeTruthy();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should handle nested Options child Somes', () => {
        const shouldRender = true;
        const shouldRenderChildOption = true
        const { container: renderedOption, asFragment } = render(
            <Option value={ shouldRender }>
              <Some>
                  <div id="some">
                      <p>I am some content</p>
                  </div>
                  <Option value={ shouldRenderChildOption }>
                      <Some>
                        <div id="some-child">
                          <p>I am some child content</p>
                        </div>
                      </Some>
                      <None>
                        <div id="none-child">
                          <p>I am none child content</p>
                        </div>
                      </None>
                  </Option>
              </Some>
              <None>
                  <div id="none">
                      <p>I am none content</p>
                  </div>
              </None>
            </Option>
        )

        const renderedSome = renderedOption.querySelector('#some');
        const renderedSomeChild = renderedOption.querySelector('#some-child');
        const renderedNone = renderedOption.querySelector('#none');
        const renderedNoneChild = renderedOption.querySelector('#none-child');

        expect(renderedSome).toBeTruthy();
        expect(renderedNone).toBeFalsy();
        expect(renderedSomeChild).toBeTruthy();
        expect(renderedNoneChild).toBeFalsy();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should handle nested Options multiple None', () => {
        const shouldRender = false;
        const shouldRenderChildOption = false
        const { container: renderedOption, asFragment } = render(
            <Option value={ shouldRender }>
              <Some>
                  <div id="some">
                      <p>I am some content</p>
                  </div>
              </Some>
              <None>
                  <div id="none">
                      <p>I am none content</p>
                  </div>
                  <Option value={ shouldRenderChildOption }>
                      <Some>
                        <div id="some-child">
                          <p>I am some child content</p>
                        </div>
                      </Some>
                      <None>
                        <div id="none-child">
                          <p>I am none child content</p>
                        </div>
                      </None>
                  </Option>
              </None>
            </Option>
        )

        const renderedSome = renderedOption.querySelector('#some');
        const renderedSomeChild = renderedOption.querySelector('#some-child');
        const renderedNone = renderedOption.querySelector('#none');
        const renderedNoneChild = renderedOption.querySelector('#none-child');

        expect(renderedSome).toBeFalsy();
        expect(renderedNone).toBeTruthy();
        expect(renderedSomeChild).toBeFalsy();
        expect(renderedNoneChild).toBeTruthy();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should handle nested Options None in Some', () => {
        const shouldRender = true;
        const shouldRenderChildOption = false
        const { container: renderedOption, asFragment } = render(
            <Option value={ shouldRender }>
              <Some>
                  <div id="some">
                      <p>I am some content</p>
                  </div>
                  <Option value={ shouldRenderChildOption }>
                      <Some>
                        <div id="some-child">
                          <p>I am some child content</p>
                        </div>
                      </Some>
                      <None>
                        <div id="none-child">
                          <p>I am none child content</p>
                        </div>
                      </None>
                  </Option>
              </Some>
              <None>
                  <div id="none">
                      <p>I am none content</p>
                  </div>
              </None>
            </Option>
        )

        const renderedSome = renderedOption.querySelector('#some');
        const renderedSomeChild = renderedOption.querySelector('#some-child');
        const renderedNone = renderedOption.querySelector('#none');
        const renderedNoneChild = renderedOption.querySelector('#none-child');

        expect(renderedSome).toBeTruthy();
        expect(renderedNone).toBeFalsy();
        expect(renderedSomeChild).toBeFalsy();
        expect(renderedNoneChild).toBeTruthy();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should handle nested Options Some in None', () => {
        const shouldRender = false;
        const shouldRenderChildOption = true
        const { container: renderedOption, asFragment } = render(
            <Option value={ shouldRender }>
              <Some>
                  <div id="some">
                      <p>I am some content</p>
                  </div>
              </Some>
              <None>
                  <div id="none">
                      <p>I am none content</p>
                  </div>
                  <Option value={ shouldRenderChildOption }>
                      <Some>
                        <div id="some-child">
                          <p>I am some child content</p>
                        </div>
                      </Some>
                      <None>
                        <div id="none-child">
                          <p>I am none child content</p>
                        </div>
                      </None>
                  </Option>
              </None>
            </Option>
        )

        const renderedSome = renderedOption.querySelector('#some');
        const renderedSomeChild = renderedOption.querySelector('#some-child');
        const renderedNone = renderedOption.querySelector('#none');
        const renderedNoneChild = renderedOption.querySelector('#none-child');

        expect(renderedSome).toBeFalsy();
        expect(renderedNone).toBeTruthy();
        expect(renderedSomeChild).toBeTruthy();
        expect(renderedNoneChild).toBeFalsy();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render nothing if no None and value evalutes to `false`', () => {
        const shouldRender = false;
        const { container: renderedOption, asFragment } = render(
            <Option value={ shouldRender }>
              <Some>
                  <div id="some">
                      <p>I am some content</p>
                  </div>
              </Some>
            </Option>
        )

        const renderedSome = renderedOption.querySelector('#some');
        const renderedNone = renderedOption.querySelector('#none');

        expect(renderedSome).toBeFalsy();
        expect(renderedNone).toBeFalsy();

        expect(asFragment()).toMatchSnapshot();
    });

    it('should render nothing if no Some and value evalutes to `true`', () => {
        const shouldRender = true;
        const { container: renderedOption, asFragment } = render(
            <Option value={ shouldRender }>
              <None>
                <div id="none">
                  <p>I am none component</p>
                </div>
              </None>
            </Option>
        )

        const renderedSome = renderedOption.querySelector('#some');
        const renderedNone = renderedOption.querySelector('#none');

        expect(renderedSome).toBeFalsy();
        expect(renderedNone).toBeFalsy();

        expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot when value is `true`', () => {
      const shouldRender = true;
      const { asFragment } = render(
        <Option value={ shouldRender }>
          <Some>
              <div id="some">
                  <p>I am some content</p>
              </div>
          </Some>
          <None>
              <div id="none">
                  <p>I am none content</p>
              </div>
          </None>
        </Option>
      );

      const { firstChild: renderedOption } = asFragment();

      expect(renderedOption).toMatchSnapshot();
    });


    it('should match snapshot when value is `false`', () => {
      const shouldRender = false;
      const { asFragment } = render(
        <Option value={ shouldRender }>
          <Some>
              <div id="some">
                  <p>I am some content</p>
              </div>
          </Some>
          <None>
              <div id="none">
                  <p>I am none content</p>
              </div>
          </None>
        </Option>
      );

      const { firstChild: renderedOption } = asFragment();

      expect(renderedOption).toMatchSnapshot();
    });
});
