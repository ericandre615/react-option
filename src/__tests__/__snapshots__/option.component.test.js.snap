// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option Component should handle nested Options None in Some 1`] = `
<DocumentFragment>
  <div
    id="some"
  >
    <p>
      I am some content
    </p>
  </div>
  <div
    id="none-child"
  >
    <p>
      I am none child content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Option Component should handle nested Options Some in None 1`] = `
<DocumentFragment>
  <div
    id="none"
  >
    <p>
      I am none content
    </p>
  </div>
  <div
    id="some-child"
  >
    <p>
      I am some child content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Option Component should handle nested Options child Somes 1`] = `
<DocumentFragment>
  <div
    id="some"
  >
    <p>
      I am some content
    </p>
  </div>
  <div
    id="some-child"
  >
    <p>
      I am some child content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Option Component should handle nested Options multiple None 1`] = `
<DocumentFragment>
  <div
    id="none"
  >
    <p>
      I am none content
    </p>
  </div>
  <div
    id="none-child"
  >
    <p>
      I am none child content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Option Component should match snapshot when value is \`false\` 1`] = `
<div
  id="none"
>
  <p>
    I am none content
  </p>
</div>
`;

exports[`Option Component should match snapshot when value is \`true\` 1`] = `
<div
  id="some"
>
  <p>
    I am some content
  </p>
</div>
`;

exports[`Option Component should render nothing if no None and value evalutes to \`false\` 1`] = `<DocumentFragment />`;

exports[`Option Component should render nothing if no Some and value evalutes to \`true\` 1`] = `<DocumentFragment />`;

exports[`Option Component should return children of None when value prop evaluates to \`false\` 1`] = `
<DocumentFragment>
  <div
    id="none"
  >
    <p>
      I am none content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Option Component should return children of Some when value prop evaluates to \`true\` 1`] = `
<DocumentFragment>
  <div
    id="some"
  >
    <p>
      I am some content
    </p>
  </div>
</DocumentFragment>
`;
